flask_sqlalchemy/__init__.py

```diff
@@ -1017,6 +1024,22 @@ def __repr__(self):
             app and app.config['SQLALCHEMY_DATABASE_URI'] or None
         )
 
+    @property
+    def bases(self):
+        return [self.Model] + self.external_bases
+
+    def register_base(self, Base):
+        """Register an external raw SQLAlchemy declarative base. 
+        Allows usage of the base with our session management and 
+        adds convenience query property using BaseQuery by default."""
+
+        self.external_bases.append(Base)
+        for c in Base._decl_class_registry.values():
+            if isinstance(c, type):
+                if not hasattr(c, 'query') and not hasattr(c, 'query_class'):
+                    c.query_class = BaseQuery
```

Comment:
Use `self.Query` instead of just BaseQuery to be consistent if a custom query class is used.